lib/internal/errors.js

```diff
@@ -286,6 +286,45 @@ function uvException(ctx) {
   return err;
 }
 
+/**
+ * This creates an error compatible with errors produced in the C++
+ * This function should replace the deprecated
+ * `exceptionWithHostPort(err, syscall, address, port, additional)` function.
+ *
+ * @param {number} err - A libuv error number
+ * @param {string} syscall
+ * @param {string} address
+ * @param {number} [port]
+ * @param {string} [additional]
+ * @returns {Error}
+ */
+function uvExceptionWithHostPort(err, syscall, address, port, additional) {
+  const [ code, uvmsg ] = errmap.get(err);
+  const message = `${syscall} ${code}: ${uvmsg}`;
+  let details = '';
+
+  if (port && port > 0) {
+    details = ` ${address}:${port}`;
+  } else if (address) {
+    details = ` ${address}`;
+  }
+  if (additional) {
+    details += ` - Local (${additional})`;
+  }
+
+  // eslint-disable-next-line no-restricted-syntax
```

Comment:
Why is this needed?

Reply:
In case of `additional` param is not null, appending the `additional` string into the `details` string.
Backward compatibility, same logic like in `exceptionWithHostPort` function.

Reply:
Huh? I'm referring to the eslint comment. I don't understand why we need to disable the rule for the `new Error()`.

Reply:
@mscdex The linter disables `new Error()` in the code base so we can enforce using the errors defined with codes in this file

Reply:
Oh, sory, i thought you ment for the `if(additinal)` statement.